src/Bottle/provider.js

```diff
@@ -42,6 +42,8 @@ var createProvider = function createProvider(name, Provider) {
         enumerable : true,
         get : function getProvider() {
             var instance = new Provider();
+            if (Array.isArray(instance.$get)) instance.$get = instance.$get[instance.$get.length - 1]
```

Comment:
`Provider.prototype.$get` should never be defined as an array. It should be a function that behaves like a factory, returning the service instance.

Can you add a unit test that shows the issue failing and how this code fixes it? 

I added a comment to your issue #187 .  Your example case seems to work fine for me when I fix the typo.

Thanks!